lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -24,19 +24,25 @@ use util::ser::{Writeable, Writer, Readable};
 use std::io::Error;
 use ln::msgs::DecodeError;
 
-/// Enforces some rules on ChannelKeys calls. Eventually we will probably want to expose a variant
-/// of this which would essentially be what you'd want to run on a hardware wallet.
+/// An implementation of ChannelKeys that enforces some policy checks.
+///
+/// NOTE: this implementation trusts the caller's pre-built Bitcoin transaction
```

Comment:
Added this new doc.  I wonder if there's a better way to make users of this struct aware of this.  Perhaps a policy enum that decides whether to rebuild the transaction?  (`InternalSignerPolicy` vs `ExternalSignerPolicy`)